<launch>

  <!-- Export the env variables in the .bashrc!! -->
  <arg name="nao_ip"            default="$(optenv NAO_IP 127.0.0.1)" />
  <arg name="nao_port"          default="$(optenv NAO_PORT 9559)" />
  <arg name="roscore_ip"        default="$(optenv ROSCORE_IP 127.0.0.1)" />
  <arg name="network_interface" default="$(optenv NETWORK_INTERFACE wlan0)" />
  <arg name="namespace"         default="start_pepper" />
  <arg name="node_start_delay"  default="10.0" />

  <!-- Load env variables to the parameter server -->
  <param name="nao_ip"                value="$(arg nao_ip)" />
  <param name="nao_port"              value="$(arg nao_port)" />
  <param name="roscore_ip"            value="$(arg roscore_ip)" />
  <param name="network_interface"     value="$(arg network_interface)" />


  <!-- Include Bringup Launch -->
  <include file="$(find pepper_bringup)/launch/pepper_full.launch">
    <arg name="nao_ip"            value="$(arg nao_ip)" />
    <arg name="roscore_ip"        value="$(arg roscore_ip)" />
    <arg name="network_interface" value="$(arg network_interface)" />
  </include>

  <!-- Launch start_pepper node to turn off autonomous life after $(arg node_start_delay) seconds-->
  <node pkg="pepper_developed" type="start_pepper" name="$(arg namespace)" launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' " args="--qi-url=tcp://$(arg nao_ip):$(arg nao_port) --roscore_ip=$(arg roscore_ip) --network_interface=$(arg network_interface) --namespace=$(arg namespace)" output="screen" />

  <!-- Launch RViz -->
  <!-- <node pkg="rviz" type="rviz" name="rviz" args="-d $(find pepper_bringup)/config/pepper.rviz" /> -->

</launch>
